application/features/email.feature

```diff
@@ -65,3 +65,154 @@ Feature: Email
+    When I check if a lost security key email should be sent
+    Then I see that a lost security key email <shouldOrNot> be sent
+
+    Examples:
+      | sendLostKeyEml | u2fUsedDaysAgo | backupUsedDaysAgo | shouldOrNot |
+      | to send        | 4              | 2                 | should NOT  |
+      | to send        | 444            | 222               | should NOT  |
+      | to send        | 444            | 2                 | should      |
+      | NOT to send    | 444            | 2                 | should NOT  |
+
+  Scenario Outline: When to send get backup codes emails
+    Given we are configured <sendGetBackupCodesEml> get backup codes emails
+      And no mfas exist
+      And a user already exists
+      And a u2f mfa option <u2fExistsOrNot>
+      And a totp mfa option <totpExistsOrNot>
+      And a backup code mfa option <backupExistsOrNot>
+    When I check if a get backup codes email should be sent
+    Then I see that a get backup codes email <shouldOrNot> be sent
+
+    Examples:
+      | sendGetBackupCodesEml | u2fExistsOrNot | totpExistsOrNot | backupExistsOrNot | shouldOrNot |
+      | to send               | does NOT exist | does NOT exist  | does NOT exist    | should NOT  |
+      | to send               | does exist     | does NOT exist  | does NOT exist    | should      |
+      | to send               | does NOT exist | does exist      | does NOT exist    | should      |
+      | to send               | does exist     | does exist      | does NOT exist    | should NOT  |
+      | to send               | does exist     | does exist      | does exist        | should NOT  |
+      | to send               | does NOT exist | does exist      | does exist        | should NOT  |
+      | to send               | does exist     | does NOT exist  | does exist        | should NOT  |
+      | to send               | does NOT exist | does NOT exist  | does exist        | should NOT  |
+      | NOT to send           | does exist     | does exist      | does NOT exist    | should NOT  |
+
+  Scenario Outline:  When to send refresh backup codes emails
+    Given we are configured <sendRefreshBackupCodesEml> refresh backup codes emails
+      And no mfas exist
+      And a user already exists
+      And a backup code mfa option <backupExistsOrNot>
+      And there are <backupCodes> backup codes
+    When I check if a refresh backup codes email should be sent
+    Then I see that a refresh backup codes email <shouldOrNot> be sent
+
+    Examples:
+      | sendRefreshBackupCodesEml | backupExistsOrNot | backupCodes | shouldOrNot |
+      | to send                   | does NOT exist    | 0           | should NOT  |
+      | to send                   | does NOT exist    | 0           | should NOT  |
+      | to send                   | does exist        | 3           | should      |
+      | to send                   | does exist        | 4           | should NOT  |
```

Comment:
Based on the comment by [`MINIMUM_BACKUP_CODES_BEFORE_NAG`](https://github.com/silinternational/idp-id-broker/pull/115/files#diff-15c4448209de48489ad86a814d4a4dbcR36), it sounds like we should send the refresh-backup-codes email when the user has 4 backup codes left, but the tests say to send when they have 3 codes left. Was that a typo in the test, or is there a better name for `MINIMUM_BACKUP_CODES_BEFORE_NAG` to make it clear whether it means `=` or `<=`?

Reply:
Clarified the comment (i.e. 4 is OK, 3 triggers a nag)